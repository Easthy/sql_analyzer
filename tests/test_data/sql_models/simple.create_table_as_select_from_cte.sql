CREATE TABLE simple.create_table_as_select_from_cte AS
WITH _cte AS (
       SELECT users.id AS user_id,
              users.name
         FROM source.users
)
SELECT _cte.user_id,
       _cte.name
  FROM _cte
;