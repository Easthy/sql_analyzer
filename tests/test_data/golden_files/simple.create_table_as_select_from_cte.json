{
  "nodes": [
    { "id": "col:simple.create_table_as_select_from_cte.name" },
    { "id": "col:simple.create_table_as_select_from_cte.user_id" },
    { "id": "tbl:simple.create_table_as_select_from_cte" },
    { "id": "tbl:source.users" },
    { "id": "col:source.users.name" },
    { "id": "col:source.users.id" }
  ],
  "links": [
    { "source": "tbl:simple.create_table_as_select_from_cte", "target": "col:simple.create_table_as_select_from_cte.name", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_from_cte", "target": "col:simple.create_table_as_select_from_cte.user_id", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_from_cte", "target": "tbl:source.users", "type": "table_dependency" },
    { "source": "tbl:source.users", "target": "col:source.users.name", "type": "contains_column" },
    { "source": "tbl:source.users", "target": "col:source.users.id", "type": "contains_column" },
    { "source": "col:simple.create_table_as_select_from_cte.name", "target": "col:source.users.name", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_cte.user_id", "target": "col:source.users.id", "type": "column_dependency" }
  ]
}